terraform/utils.go

```diff
@@ -0,0 +1,102 @@
+package terraform
+
+import (
+	"sort"
+	"strings"
+)
+
+func isTerraformTfvarsFile(fileName string) bool {
+	// the terraform.tfvars file is a strict file name so make sure the file isn't called something like *terraform.tfvars
+	return fileName == DEFAULT_TFVARS || strings.HasSuffix(fileName, "/"+DEFAULT_TFVARS)
+}
+
+func isValidVariableFile(fileName string) bool {
+	if isTerraformTfvarsFile(fileName) {
+		return true
+	}
+	for _, fileExt := range VALID_VARIABLE_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidTerraformFile(fileName string) bool {
+	for _, fileExt := range VALID_TERRAFORM_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+type PrioritisableFile struct {
+	fileName string
+	priority int
+}
+
+func createPrioritisableFile(fileName string) PrioritisableFile {
+	// TODO: Variables in terraform.tfvars.json files come after terraform.tfvars and before .auto.tfvars (not supported)
+	// TODO: Variables in -var and -var-file options come after .auto.tfvars and .auto.tfvars.json files, in the order they are provided (not supported)
+	if strings.HasSuffix(fileName, TF) {
+		// Default values have lowest priority (in .tf files)
+		return PrioritisableFile{
+			fileName: fileName,
+			priority: 3,
+		}
+	} else if isTerraformTfvarsFile(fileName) {
+		// Then variables in the terraform.tfvars file if it exists
+		return PrioritisableFile{
+			fileName: fileName,
+			priority: 2,
+		}
+	} else if strings.HasSuffix(fileName, AUTO_TFVARS) {
+		// Then variables in .auto.tfvars, in lexical order
+		// TODO: or.auto.tfvars.json (not supported)
+		return PrioritisableFile{
+			fileName: fileName,
+			priority: 1,
+		}
+	} else {
+		// Won't happen
+		return PrioritisableFile{
+			fileName: fileName,
+			priority: 0,
```

Comment:
I love that you took care of this situation even though it will not happen. :)
One last feedback, reversing the priority numbering (so lowest priority will be 1) will make more sense and it will make adding new file types with higher priority more easy (currently we will need to add 1 to each file type priority to free the number "1" for the new file type).

Reply:
Turns out the higher the value the higher the priority so your suggestion is more correct 👍